File: DiagonalDifference.go
package main

import "fmt"

func diagonalDifference(arr [][]int32) int32 {
	var first,second int32
	first = 0
	second = 0
	arrSize := len(arr) - 1
	for i := 0; i <= arrSize; i++ {
		for j := 0; j <= arrSize; j++ {
			if i == j {
				first += arr[i][j]
				second += arr[i][arrSize-j]
			}
		}
	}
	result := first-second
	if result < 0 {
		result *= -1
	}
	return result
}

func main() {
	arr := [][]int32{{1, 2, 3}, {4, 5, 6}, {1, 8, 9}}
	fmt.Println(diagonalDifference(arr))
}

File: DiagonalDifference.php
<?php

function diagonalDifference($arr)
{
    $arraySize = sizeOf($arr) - 1;
    $firstDiagonal = 0;
    $secondDiagonal = 0;
    for ($i = 0; $i <= $arraySize; $i++) {
        for ($j = 0; $j <= $arraySize; $j++) {
            if ($i == $j) {
                $firstDiagonal += $arr[$i][$j];
                $secondDiagonal += $arr[$i][$arraySize - $j];
            }
        }
    }
    $result = $firstDiagonal - $secondDiagonal;
    if ($result < 0) {
        $result *= -1;
    }
    return $result;
}

$arr = [
    [1, 2, 3],
    [4, 5, 6],
    [1, 8, 9]
];

var_dump(diagonalDifference($arr));
